lib/kubernetes-deploy/kubernetes_resource/replica_set.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class ReplicaSet < KubernetesResource
+    TIMEOUT = 5.minutes
+
+    def initialize(namespace:, context:, definition:, logger:, parent: nil, deploy_started: nil)
+      @parent = parent
+      @deploy_started = deploy_started
+      @rollout_data = { "replicas" => 0 }
+      @pods = []
+      super(namespace: namespace, context: context, definition: definition, logger: logger)
+    end
+
+    def sync(rs_data = nil)
+      if rs_data.blank?
+        raw_json, _err, st = kubectl.run("get", type, @name, "--output=json")
+        rs_data = JSON.parse(raw_json) if st.success?
+      end
+
+      if rs_data.present?
+        @found = true
+        @rollout_data = { "replicas" => 0 }.merge(rs_data["status"]
+          .slice("replicas", "availableReplicas", "readyReplicas"))
+        @status = @rollout_data.map { |state_replicas, num| "#{num} #{state_replicas.chop.pluralize(num)}" }.join(", ")
+        @pods = find_pods(rs_data)
+      else # reset
+        @found = false
+        @rollout_data = { "replicas" => 0 }
+        @status = nil
+        @pods = []
+      end
+    end
+
+    def deploy_succeeded?
+      @rollout_data["replicas"].to_i == @rollout_data["availableReplicas"].to_i &&
+      @rollout_data["replicas"].to_i == @rollout_data["readyReplicas"].to_i
+    end
+
+    def deploy_failed?
+      @pods.present? && @pods.all?(&:deploy_failed?)
+    end
+
+    def deploy_timed_out?
+      super || @pods.present? && @pods.all?(&:deploy_timed_out?)
+    end
+
+    def exists?
+      @found
+    end
+
+    def desired_replicas
+      @definition["spec"]["replicas"].to_i
+    end
+
+    def fetch_events
+      own_events = super
+      return own_events unless @pods.present?
+      own_events.merge(@pods.first.fetch_events)
```

Comment:
pods.first: does that mean we'll lose some data or is it the same for the entire thing?